src/util/Util.ts

```diff
@@ -60,19 +60,24 @@ export const matchingMediaType = (mediaA: string, mediaB: string): boolean => {
 
 /**
  * Pipes one stream into another.
- * Makes sure an error of the first stream gets passed to the second.
+ * In case of an error in the first stream the second one will be destroyed with the given error.
```

Comment:
Why not the first too, though?

Reply:
I didn't test extensively but I interpreted the node docs as this only being an issue when the first stream errors, which would cause the second stream to keep waiting (since there was never an end event, only an error event probably?). If the second one errors and closes the pipe would see that there is nothing to pipe to anymore. But this is all conjecture based on the documentation, will see if it's possible to do some more tests there without having to delp into the node code itself.